source/usage-examples/transaction-core.txt

```diff
@@ -0,0 +1,239 @@
+
+.. code-block:: json
+
+   { _id: 98765, orders: [] }
+
+The following documents are in the ``inventory`` collection:
+
+.. code-block:: json
+
+   { item: "sunblock", item_id: 5432, qty: 85 },
+   { item: "beach towel", item_id: 7865, qty: 41 }
+
+You store purchase records in the ``orders`` collection of the
+``testdb`` database. This collection is empty, as there have been no
+purchases.
+
+The code examples use the ``cart`` and ``payment`` variables to represent
+a sample list of items purchased and the order payment details. The
+following code describes the contents of the ``cart`` and ``payment`` variables:
+
+.. code-block:: javascript
+
+   const cart = [
+     { item: 'sunblock', item_id: 5432, qty: 1, price: 5.19 },
+     { item: 'beach towel', item_id: 7865, qty: 2, price: 15.99 }
+   ];
+   const payment = { customer: 98765, total: 37.17 };
+
+.. _nodejs-transaction-core-api-example:
+
+Implementation
+~~~~~~~~~~~~~~
+
+The code example in this section demonstrates how to use the Core API to
+perform a multi-document transaction in a session. In this example, the
+transaction reflects the actions needed when a customer purchases items from
+your store.
+
+This example code performs a transaction through the following actions:
+
+1. Calls the ``startSession()`` method to create a new session.
+
+#. Calls the ``startTransaction()`` method with an options parameter to
+   create a new transaction.
+
+#. Performs the following operations within the transaction:
+
+   * Inserts a document to the ``orders`` collection that contains information
+     about the customer, items purchased, and order total.
+   * Checks if there is sufficient inventory to fulfill the purchase.
+   * Returns an error message if the items are not available or there
+     are not enough items.
+   * If the inventory is sufficient for the order, the code performs an
+     update operation in the ``inventory`` collection to reduce the
+     quantities of the items purchased.
+   * Adds the ID of the order to the list of past orders for the customer.
+   * Returns a message acknowledging that the transaction
+     committed successfully with a copy of the purchase record.
+
+#. Calls the ``commitTransaction()`` method to commit the transaction if
+   all operations complete successfully.
+
+#. Implements a ``catch`` block that contains error-handling logic.
+
+#. Calls the ``abortTransaction()`` method to terminate the transaction.
+
+#. Calls the ``endSession()`` method to terminate the session.
+
+.. literalinclude:: /code-snippets/transactions/txn-core.js
+   :language: javascript
+   :linenos:
+   :emphasize-lines: 3-7,10,13,24,38,51,55,58-67,71,74
+   :start-after: start placeOrder
+   :end-before: end placeOrder
+
+.. _nodejs-transaction-example-payment-result:
+
+Transaction Results
+~~~~~~~~~~~~~~~~~~~
+
+When your application completes the payment transaction, the collections
+in the database contains the customer and order updates. 
```

Comment:
Issue:
I think there is mismatched verb agreement between "collections (in the database)" and "contains".

Suggestion:
"collections in the database contain"